File: src/java/src/main/java/org/elastos/hive/vault/scripting/OrCondition.java
package org.elastos.hive.vault.scripting;

import java.util.List;

/**
 * Vault script condition that succeeds if at least one of the contained conditions are successful.
 * Contained conditions are tested in the given order, and test stops as soon as one successful condition
 * succeeds.
 */
public class OrCondition extends AggregatedCondition {
	public OrCondition(String name, List<Condition> conditions) {
		super(name, Type.OR, conditions);
	}

	public OrCondition(String name) {
		this(name, null);
	}
}

File: src/service/scripting/orcondition.ts
import {AggregatedCondition} from "./aggregatedcondition";
import {Condition} from "./condition";

/**
 * Vault script condition that succeeds if at least one of the contained conditions are successful.
 * Contained conditions are tested in the given order, and test stops as soon as one successful condition
 * succeeds.
 */
export class OrCondition extends AggregatedCondition {
    constructor(name: string, conditions?: Condition[]) {
        super(name, AggregatedCondition.OR, conditions);
    }
}
